components/public-api-server/pkg/apiv1/oidc.go

```diff
@@ -318,3 +326,28 @@ func toDbOIDCSpec(oauth2Config *v1.OAuth2Config, oidcConfig *v1.OIDCConfig) db.O
 		Scopes:       append([]string{goidc.ScopeOpenID, "profile", "email"}, oauth2Config.GetScopes()...),
 	}
 }
+
+func assertIssuerIsReachable(host string) error {
```

Comment:
This should accept the `context` and construct the request with context (`http.NewRequest().WithContext()`). This ensures that if the caller cancels, we also cancel.

Reply:
In general, anything crossing a network boundary should accept context, and propagate it.